mapiproxy/servers/default/emsmdb/emsmdbp_object.c

```diff
@@ -2631,7 +2631,10 @@ static int emsmdbp_object_get_properties_mapistore_root(TALLOC_CTX *mem_ctx, str
 			}
 		}
                 else {
-			retval = openchangedb_get_folder_property(data_pointers, emsmdbp_ctx->oc_ctx, emsmdbp_ctx->username, properties->aulPropTag[i], folder->folderID, data_pointers + i);
+                        /* We are not using emsmdbp_ctx->username because we want to impersonate to get the properties
+                           on shared folders */
+			owner = emsmdbp_get_owner(object);
+			retval = openchangedb_get_folder_property(data_pointers, emsmdbp_ctx->oc_ctx, owner, properties->aulPropTag[i], folder->folderID, data_pointers + i);
```

Comment:
@sixstone-qq I thought we were able to browse public folders hierarchy created in openchangedb, but I think I have mistaken with the ability to list and click on root public folders. Therefore the public folder test shall not be relevant here.

However I would like you to make this code resilient to `owner == NULL` which was the case I expected you to test with public folders.